scvi/model/base/_base_model.py

```diff
@@ -81,6 +81,24 @@ def __init__(self, adata: Optional[AnnData] = None):
         self.history_ = None
         self._data_loader_cls = AnnDataLoader
 
+    @property
+    def adata(self) -> AnnData:
+        """Data attached to model instance."""
+        return self._adata
+
+    @adata.setter
+    def adata(self, adata: AnnData):
+        if adata is None:
+            raise ValueError("adata cannot be None.")
+        self._validate_anndata(adata)
+        self._adata = adata
+        self._adata_manager = self.get_anndata_manager(adata)
```

Comment:
This needs to reset self.registry_ and self.summary_stats_. Should have a test corresponding to this as well.

Reply:
Would add `required=True` here since we never want None returned here (even if validate shouldnt let that happen anyways)

Reply:
I will add this

Reply:
still need required flag